homeassistant/components/life360/config_flow.py

```diff
@@ -1,108 +1,251 @@
+
+    async def _async_verify(self, step_id: str) -> FlowResult:
+        """Attempt to authorize the provided credentials."""
+
+        assert self._api
+        assert self._username
+        assert self._password
+
+        errors: dict[str, str] = {}
+        authorization = await get_life360_authorization(
+            self.hass, self._api, self._username, self._password, errors
+        )
+        if errors:
+            if step_id == "user":
+                schema = account_schema(
+                    self._username, self._password
+                ) | _account_options_schema(self._options)
             else:
-                if self._username in self.configured_usernames:
-                    errors["base"] = "already_configured"
-                else:
-                    return self.async_create_entry(
-                        title=self._username,
-                        data={
-                            CONF_USERNAME: self._username,
-                            CONF_PASSWORD: self._password,
-                            CONF_AUTHORIZATION: authorization,
-                        },
-                        description_placeholders={"docs_url": DOCS_URL},
-                    )
-
-        data_schema = OrderedDict()
-        data_schema[vol.Required(CONF_USERNAME, default=self._username)] = str
-        data_schema[vol.Required(CONF_PASSWORD, default=self._password)] = str
+                # Don't show current password the first time we prompt for password
+                # since this will happen asynchronously. However, once the user enters a
+                # password, we can show it in case it's not valid to make it easier to
+                # enter a long, complicated password.
+                pwd = vol.UNDEFINED if self._first_reauth_confirm else self._password
+                self._first_reauth_confirm = False
+                schema = password_schema(pwd)
+            return self.async_show_form(
+                step_id=step_id, data_schema=vol.Schema(schema), errors=errors
+            )
 
-        return self.async_show_form(
-            step_id="user",
-            data_schema=vol.Schema(data_schema),
-            errors=errors,
-            description_placeholders={"docs_url": DOCS_URL},
+        data = {
+            CONF_USERNAME: self._username,
+            CONF_PASSWORD: self._password,
+            CONF_AUTHORIZATION: authorization,
+        }
+
+        if self._reauth_entry:
+            LOGGER.info("Reauthorization successful")
+            self.hass.config_entries.async_update_entry(self._reauth_entry, data=data)
+            if self._reauth_entry.state == ConfigEntryState.LOADED:
+                # Config entry reload should not be necessary. Restarting coordinator's
+                # scheduled refreshes should be sufficient since Life360 api object is
+                # valid again after successful reauthorization.
+                coordinator = self.hass.data[DOMAIN]["accounts"][self.unique_id][
```

Comment:
We should always create a new api instance when setting up the config entry. The config flow should not access hass.data. The config flow creates a result and that is used to create or update the config entry. Then we set up the integration with the config entry.

Reply:
Still don't understand why. If there's data that can be, or needs to be, shared between the config flow and main flow, what's wrong with storing it in hass.data? I know I'm not supposed to use global data for that.

Reply:
Ok, removed all references to hass.data from config_flow.py.